### PR TITLE
Rename qro(a)m target registers to avoid name collision with named qubits

### PR DESCRIPTION
**Pull request by fdmalone**
If the number of data sets is >= 10 collisions could occur (see #556).

### CODE REVIEW

**Comment by fdmalone** [comment]
also a hot fix for a flaky THC test.

File: qualtran/bloqs/select_swap_qrom.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def selection_registers(self) -> Tuple[SelectionRegister, ...]:
     @cached_property
     def target_registers(self) -> Tuple[Register, ...]:
         return tuple(
-            Register(f'target{sequence_id}', self._target_bitsizes[sequence_id])
+            Register(f'target{sequence_id}_', self._target_bitsizes[sequence_id])
```
Comment (mpharrigan) [inline qualtran/bloqs/select_swap_qrom.py:154]: add a comment explaining why the underscore is necessary and link to issue

**Review by mpharrigan — APPROVED** [review]
let's leave the issue open so we don't forget about this kludge

File: qualtran/bloqs/select_swap_qrom.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/select_swap_qrom.py:154]: Done.

**Pull request merged**